src/mofaflex/_core/model.py

```diff
@@ -144,6 +175,11 @@ def _get_plates(self, subsample=None):
                 dim=self._feature_plate_dim,
             )
 
+        for guiding_var_name in self.guiding_vars_names:
+            plates[f"guiding_vars_{guiding_var_name}"] = pyro.plate(
+                f"plate_guiding_vars_{guiding_var_name}", 1, subsample=torch.arange(1), dim=self._feature_plate_dim
+            )
+
```

Comment:
Since it's always the same dimension, it should be possible to use the same plate for all guiding vars.